app/controllers/v0/map_services_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'map/security_token/service'
+
+module V0
+  class MapServicesController < SignIn::ApplicationController
+    skip_before_action :authenticate
+    before_action :load_service_account_config
+
+    # POST /v0/map_services/token
+    def token
+      result = MAP::SecurityToken::Service.new.token(
+        application: @service_account_config.service_account_id.to_sym,
+        icn: params[:icn]
+      )
+
+      render json: result, status: :ok
+    rescue Common::Client::Errors::ClientError
+      render json: sts_client_error, status: :bad_gateway
+    rescue MAP::SecurityToken::Errors::ApplicationMismatchError
+      render json: application_mismatch_error, status: :bad_request
+    end
+
+    private
+
+    def sts_client_error
+      {
+        error: 'server_error',
+        error_description: "STS failed to return a valid token. Request ID: #{request.request_id}"
+      }
+    end
+
+    def application_mismatch_error
+      {
+        error: 'invalid_request',
+        error_description: "Application mismatch detected. Application: #{@service_account_config.service_account_id}"
```

Comment:
I probably wouldn't worry about adding the service_account_id in the error response here, we'll already know from the logs what the attempted application was, and the caller will have already used `service_account_id` when creating a JWT to authenticate anyways